ScrapPackedExplorerCli/CliApp.cs

```diff
@@ -61,34 +77,82 @@ private int RunExtract(ExtractOptions options)
                 ScrapPackedFile packedFile = new ScrapPackedFile(options.packedFile);
                 packedFile.Extract(options.packedPath, options.destinationPath);
             }
-            catch (Exception ex) { return Error(ex); }
+            catch (Exception ex)
+            {
+                Console.Error.WriteLine($"Error: {ex.Message}");
+                return 1;
+            }
 
             return 0;
         }
 
         private int RunList(ListOptions options)
         {
-            try {
-                ScrapPackedFile packedFile = new ScrapPackedFile(options.packedFile);            
-                List<string> fileNames = packedFile.GetFileNames();
+            try
+            {
+                ScrapPackedFile packedFile = new ScrapPackedFile(options.packedFile);
+                List<string> FileList = packedFile.GetFileNames();
+                FileList.Sort();
 
-                if (fileNames.Count == 0)
-                    Console.WriteLine($"{options.packedFile} is empty.");
+                if (FileList.Count == 0)
+                    Console.WriteLine($"'{options.packedFile}' is empty.");
                 else
-                    foreach (var fileName in fileNames)
-                        Console.WriteLine(fileName);
-            }
-            catch (Exception ex) { return Error(ex); }
+                {
+                    string query = options.searchString;
+                    if (!options.isRegex)
+                        query = Regex.Escape(query);
 
-            // Todo: implement RunList output styles
-            return 0;
-        }
+                    query = query.Replace("/", @"\/");
+                    query = query.Replace("\\*", ".*");
+                    query = query.Replace("\\?", ".");
 
-        // This just to make code "prettier". Multi-line `catch` with one-line `try` is kinda ugly
-        private int Error(Exception ex)
-        {
-            Console.Error.WriteLine($"Error: {ex.Message}");
-            return 1;
+                    if (options.StartsWith)
+                        query = "^" + query;
+
+                    Regex rg = new Regex(query);
+
+                    bool found = false;
+                    foreach (var File in FileList)
+                    {
+                        OutputStyles Styles = options.outputStyle;
+
+                        var FileData = File.Split("\t");
+                        string FilePath = Path.GetDirectoryName(FileData[0]).Replace("\\", "/");
+                        string FileName = Path.GetFileName(FileData[0]);
+                        string FileSize = FileData[1];
+                        string FileOffset = FileData[2];
```

Comment:
why not iterate through FileIndexDataList? that has PackedFileIndexData with properties for filename filesize and fileoffset …
ok we might have to make the list sortable* (just don't sort the original list)
*add comparer to PackedFileIndexData

Reply:
Not so familiar with PackedFileIndexData yet, any way wanna chage it some day